Dfa feature 3

ShahParth12 opened the pull request.

ShahParth12 committed:
Added NFA and it's functionality

DevUt commented on lib/NFA/creation/nfa_create_manual.dart:
```diff
@@ -0,0 +1,133 @@
+        child: Column(
+          children: [
+            const SizedBox(
+              width: 100,
```
Why give it a width?

DevUt commented on lib/NFA/options/option.dart:
```diff
@@ -0,0 +1,377 @@
+import 'package:flutter/material.dart';
+
+class OptionsMenu extends StatefulWidget {
+  NFA NFAobj;
```
Use lowerCamelCase

DevUt commented on lib/NFA/options/option.dart:
```diff
@@ -0,0 +1,377 @@
+}
+
+class _OptionsMenuState extends State<OptionsMenu> {
+  bool test_enableflag = false;
```
Same here

DevUt commented on lib/NFA/options/option.dart:
```diff
@@ -0,0 +1,377 @@
+    TextEditingController test_outstate = TextEditingController();
+
+    TextEditingController step_readinp = TextEditingController();
+    TextEditingController step_setout = TextEditingController();
```
same

DevUt commented on lib/NFA/options/option.dart:
```diff
@@ -0,0 +1,377 @@
+                  leading: Icon(Icons.settings),
+                  title: Text('Each stage transitions'),
+                  subtitle:
+                      Text("returns step wise transitions for each alphabet"),
```
const modifier everywhere pls

DevUt committed:
Transition Table: Replace the proceed elevated with Icon Button

DevUt committed:
NFA: Add support for epsilon in transition function

DevUt committed:
NFA: Add testing input on NFA

We had stepwise input check, this allows to just get the final state and
whether it got accepted or rejected.

DevUt reviewed (approved):
LGTM
